cosmwasm/contracts/yield-ward/src/execute/response_unstake.rs

```diff
@@ -71,20 +74,22 @@ pub fn try_handle_unstake_response(
         // update action stage
         unstake_item.action_stage = UnstakeActionStage::Failed;
 
-        let lp_mint_msg = create_cw20_transfer_msg(
-            &token_config.lpt_address,
-            &unstake_item.user,
-            unstake_item.lp_token_amount,
-        )
-        .ok_or(ContractError::CustomError(
-            "Can't create CW20 transfer message".to_owned(),
-        ))?;
-        response = response.add_message(lp_mint_msg);
-
-        let unstake_failed_event = Event::new("unstake_failed")
-            .add_attribute("unstake_id", unstake_response.unstake_id.to_string())
-            .add_attribute("lp_amount", unstake_item.lp_token_amount.to_string());
-        events.push(unstake_failed_event);
+        response = response
+            .add_message(
+                create_cw20_transfer_msg(
+                    &token_config.lpt_address,
+                    &unstake_item.user,
+                    unstake_item.lp_token_amount,
+                )
+                .ok_or(ContractError::CustomError(
+                    "Can't create CW20 transfer message".to_owned(),
+                ))?,
+            )
+            .add_event(
+                Event::new("unstake_failed")
+                    .add_attribute("unstake_id", unstake_response.unstake_id.to_string())
+                    .add_attribute("lp_amount", unstake_item.lp_token_amount.to_string()),
+            );
     }
 
     UNSTAKES.save(
```

Comment:
You have redundant `UNSTAKES.remove` higher on line 66 which clears the same storage slot

Reply:
fixed